File: baek1920.py
n = int(input())
n_list = list(map(int, input().split(" ")))
n_list.sort()
m = int(input())
targets = list(map(int, input().split(" ")))

def binary(target):
    start = 0
    end = n - 1

    while start <= end:
        mid = (start + end) // 2

        if n_list[mid] == target:
            return 1
        
        if target < n_list[mid]:
            end = mid - 1
        else:
            start = mid + 1
    return 0

for i in targets:
    print(binary(i))

File: baek1920.swift
import Foundation

let N = Int(readLine()!)!

var arr = readLine()!.split(separator: " ").map { Int(String($0))! }
arr.sort()

var _ = Int(readLine()!)!

let find = readLine()!.split(separator: " ").map { Int(String($0))! }
var answer = ""
func binary(_ element: Int) -> Bool {
    var left = 0
    var right = N-1

    while left <= right {
        let mid = (left+right) / 2
        let value = arr[mid]

        if value == element {
            return true
        } else if value > element {
            right = mid - 1
        } else {
            left = mid + 1
        }
    }

    return false
}


find.forEach { value in 
    answer += binary(value) ? "1\n" : "0\n"
}

print(answer)